ToDoApp/ToDoApp/Respository/Respository.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  Respository.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-22.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+protocol Repository {
+    func createOrUpdate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID
+    func read<T>() async -> Result<[T], Error> where T: Decodable
+}
+extension DatabaseReference: Repository {
+  // MARK: create a task
+    func createOrUpdate<T>(_ value: T) async -> Error? where T: Identifiable, T: Encodable, T.ID == UUID {
+        guard let encodedValue = value.toDictionary else {
+            return .some("Error encoding value" as! Error)
```

Comment:
Force unwrap should not be used unless there's no other way to achieve something and you are 100% sure that it won't crash. String can not be casted to the Error type, since string does not conform to Error protocol and this code will crash. A value of a custom Error type should be used on that case like in previous example. 
```
enum RepositoryError: Error { 
    case createOrUpdateError
}

...

return RepositoryError. createOrUpdateError
```